src/REST/keywords.py

```diff
@@ -77,6 +78,45 @@ def set_client_cert(self, cert):
         self.request["cert"] = self._input_client_cert(cert)
         return self.request["cert"]
 
+
+    @keyword(name="Set Client Authentication", tags=("settings",))
+    def set_client_authentication(self, auth_type, user=None, password=None):
+        """*Attaches HTTP basic authentication to the given requests.*
+
+        The API by default will not have authentication enabled. 
+
+        The auth_type argument must be ${NONE}, basic, digest or proxy.
+        In case the user sets the auth_type to ${NONE} the authentication 
+        information 
+        The user and password will be written in plain text.
+
+        *Examples*
+
+        | `Set Client Authentication` | basic  | admin | password |
+        | `Set Client Authentication` | digest | admin | password |
+        | `Set Client Authentication` | proxy  | admin | password |
+        | `Set Client Authentication` | ${NONE} | | |
+        """
+        if auth_type != None and not isinstance(auth_type, str):
+            raise TypeError(
+                    "Argument \"auth_type\" must be string or ${NONE}!"
+            )
+
+        if auth_type != None:
```

Comment:
This isn't necessary as you've already checked if `auth_type` is None

Reply:
This is required for the case in which auth_type is None. Because it will use the lower() function will cause an error.